---
title: "Allele_Calling"
output: html_document
date: "2024-09-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)

# install wes anderson color palette
if(!require(wesanderson)){
    install.packages("wesanderson")
    library(wesanderson)
}
```

# importing the data:
```{r}
##########################################################################
#filtered data:
alleleCall_filtered <- read.delim("/blue/lien.nguyen/rossellwood/repeat_expansion_tools/sort_expansion_type/filtered_full_FASTQ/alleleCall_filtered23.txt", header=FALSE) %>% 
  mutate(pattern = ifelse(grepl("_p1\\.", V1), "p1", 
                  ifelse(grepl("_p2\\.", V1), "p2",
                  ifelse(grepl("_p3\\.", V1), "p3",
                  ifelse(grepl("_p4\\.", V1), "p4",
                  ifelse(grepl("_p5\\.", V1), "p5",
                  ifelse(grepl("_p6\\.", V1), "p6",
                  ifelse(grepl("_p7\\.", V1), "p7",
                  ifelse(grepl("_p8\\.", V1), "p8",
                  ifelse(grepl("_p9\\.", V1), "p9",
                  ifelse(grepl("_p10\\.", V1), "p10",
                  ifelse(grepl("_p11\\.", V1), "p11",
                  ifelse(grepl("_p12\\.", V1), "p12",
                  ifelse(grepl("_p13\\.", V1), "p13",
                         NA)))))))))))))) %>% 
  rename(filename = V1,
         total_reads = V2,
         allele1_hex = V3,
         allele1_hex_count = V4,
         allele1_hex_ratio = V5,
         allele1_VNTR = V6,
         allele1_VNTR_count = V7,
         allele1_full_type = V8,
         allele1_full_count = V9,
         allele1_hexhexvar = V10,
         allele1_hexhexvar_count = V11,
         allele2_hex = V12,
         allele2_hex_count = V13,
         allele2_hex_ratio = V14,
         allele2_VNTR = V15,
         allele2_VNTR_count = V16,
         allele2_hexhexvar = V17,
         allele2_hexhexvar_count = V18,
         allele2_full_type = V19,
         allele2_full_count = V20,
         shortHex_sub_allele1 = V21,
         shorthex_sub_allele1_count = V22,
         longHex_sub_allele1 = V23,
         longHex_sub_allele1_count = V24) %>%
  mutate(Category = ifelse(grepl("_Control_", filename), "Control", 
                           ifelse(grepl("_AD_", filename), "AD", NA)))

# only keep columns that are useful:
alleleCall_concise <- subset(
  alleleCall_filtered, select = c(filename, ((total_reads)), allele1_full_type, allele1_full_count, allele2_full_type, allele2_full_count,shortHex_sub_allele1, shorthex_sub_allele1_count, longHex_sub_allele1, longHex_sub_allele1_count)) %>% 
  mutate(
    allele1_full_count = (allele1_full_count / total_reads) * 100,
    allele2_full_count = (allele2_full_count / total_reads) * 100,
    shorthex_sub_allele1_count = (shorthex_sub_allele1_count / total_reads) * 100,
    longHex_sub_allele1_count = (longHex_sub_allele1_count / total_reads) * 100) %>% 
  rename(allele1_full_percent=allele1_full_count,
         allele2_full_percent=allele2_full_count,
         shorthex_sub_allele1_percent=shorthex_sub_allele1_count,
         longHex_sub_allele1_percent=longHex_sub_allele1_count) %>% 
  mutate(percent_unknown = 100 - allele1_full_percent - allele2_full_percent)

#problem files:
problem_rows <- alleleCall_concise %>% filter(percent_unknown > 50)
good_rows <- alleleCall_concise %>% filter(percent_unknown <= 50)

# getting rid of samples that only have 1 read
alleleCall_filtered <- alleleCall_filtered %>% 
  filter(!(total_reads <=1))
```

# pie charts:
```{r}
# Calculate percentages
alleleCall_percent <- alleleCall_filtered %>%
  group_by(pattern, Category, allele1_full_type) %>%
  summarize(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)  # Calculate percentage

# Create the pie chart
ggplot(alleleCall_percent, aes(x = "", y = percentage, fill = allele1_full_type)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar(theta = "y") +
  labs(title = "Pie Chart of Allele 1 Variants (Percentages)") +
  facet_wrap(. ~ pattern) +
  # scale_fill_brewer(palette = "Set1") +  # Change to your preferred palette
  theme_minimal() +  # Use a minimal theme
  theme(axis.title.x = element_blank(),  # Remove x-axis title
        axis.title.y = element_blank(),  # Remove y-axis title
        axis.ticks = element_blank(),     # Remove axis ticks
        axis.text.x = element_blank())    # Remove x-axis text

# Calculate percentages
alleleCall_percent <- alleleCall_filtered %>%
  group_by(pattern, Category , allele2_full_type) %>%
  summarize(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)  # Calculate percentage
# Create the pie chart
ggplot(alleleCall_percent, aes(x = "", y = percentage, fill = allele2_full_type)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar(theta = "y") +
  labs(title = "Pie Chart of Allele 2 Variants (Percentages)") +
  facet_wrap(. ~ pattern) +
  scale_fill_brewer(palette = "Set1") +  # Change to your preferred palette
  theme_minimal() +  # Use a minimal theme
  theme(axis.title.x = element_blank(),  # Remove x-axis title
        axis.title.y = element_blank(),  # Remove y-axis title
        axis.ticks = element_blank(),     # Remove axis ticks
        axis.text.x = element_blank())    # Remove x-axis text
##################################################################

##############################################
# Calculate percentages
alleleCall_percent <- alleleCall_filtered %>%
  group_by(pattern, Category, allele1_full_type) %>%
  summarize(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)  # Calculate percentage

# Create the pie chart
ggplot(alleleCall_percent, aes(x = "", y = percentage, fill = allele1_full_type)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar(theta = "y") +
  labs(title = "Pie Chart of Allele 1 Variants (Percentages)") +
  facet_wrap(. ~ pattern+Category) +
  scale_fill_brewer(palette = "Set1") +  # Change to your preferred palette
  theme_minimal() +  # Use a minimal theme
  theme(axis.title.x = element_blank(),  # Remove x-axis title
        axis.title.y = element_blank(),  # Remove y-axis title
        axis.ticks = element_blank(),     # Remove axis ticks
        axis.text.x = element_blank())    # Remove x-axis text

# Calculate percentages
alleleCall_percent <- alleleCall_filtered %>%
  group_by(pattern, Category , allele2_full_type) %>%
  summarize(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)  # Calculate percentage
# Create the pie chart
ggplot(alleleCall_percent, aes(x = "", y = percentage, fill = allele2_full_type)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar(theta = "y") +
  labs(title = "Pie Chart of Allele 2 Variants (Percentages)") +
  facet_wrap(. ~ pattern+Category) +
  scale_fill_brewer(palette = "Set1") +  # Change to your preferred palette
  theme_minimal() +  # Use a minimal theme
  theme(axis.title.x = element_blank(),  # Remove x-axis title
        axis.title.y = element_blank(),  # Remove y-axis title
        axis.ticks = element_blank(),     # Remove axis ticks
        axis.text.x = element_blank())    # Remove x-axis text
##################################################################


```

# Are long hex more associated with AD than control?:
```{r}

longHex_filtereDF <- alleleCall_filtered %>% select(filename, allele1_hex, allele2_hex, allele1_VNTR, allele2_VNTR,Category)
# significantly different groups?

##########################################################
# Expand the dataframe to treat each allele as a separate sample
expanded_df <- longHex_filtereDF %>%
  pivot_longer(cols = c(allele1_hex, allele2_hex, allele1_VNTR, allele2_VNTR), names_to = "allele_position", values_to = "allele_type")

#total number of ad alleles and control alleles. (divided by 2 so that i can get the proper count, since it is pivot longered)
total_AD_alleles <- expanded_df %>% filter(Category == "AD") %>% nrow()
total_AD_alleles <- (total_AD_alleles/2)
total_CTRL_alleles <- expanded_df %>% filter(Category == "Control") %>% nrow()
total_CTRL_alleles <- (total_CTRL_alleles/2)

# Filter for LongHex alleles
long_hex_df <- expanded_df %>%
  filter(allele_type == "LongHex")

######################
# Count occurrences of each category
category_counts <- long_hex_df %>%
  group_by(Category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = ifelse(Category == "AD", (count / total_AD_alleles) * 100,
                             (count / total_CTRL_alleles) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", total_AD_alleles, total_CTRL_alleles)))  # Create a fraction string

# Create a contingency table for the chi-squared test
contingency_table <- table(long_hex_df$Category)

# Perform the chi-squared test
chi_squared_result <- chisq.test(contingency_table)

# Print the results
print(contingency_table)
print(chi_squared_result)

# Create the bar chart
ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
            position = position_stack(vjust = 0.5), 
            color = "black") +  # Add percentages and fractions on top of the bars
  labs(title = "Percent of AD/CTRL alleles in all expanded Hex alleles", x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
  theme_minimal() +
  scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))
```

Similar to expanded hexamers, we can ask the question, in expanded VNTR cases, do we see more AD?

# VNTR (v2)
```{r, echo=F, fig.cap="**Figure 3.** This graph shows the percentage of expanded VNTR alleles that are AD or Control. With the percentages so close, it is difficult to make a conclusion on if AD is significanly different from Control."}
longHex_filtereDF <- alleleCall_filtered %>% select(filename, allele1_hex, allele2_hex, allele1_VNTR, allele2_VNTR,Category)
# significantly different groups?

##########################################################
# Expand the dataframe to treat each allele as a separate sample
expanded_df <- longHex_filtereDF %>%
  pivot_longer(cols = c(allele1_hex, allele2_hex, allele1_VNTR, allele2_VNTR), names_to = "allele_position", values_to = "allele_type")

#total number of ad alleles and control alleles. (divided by 2 so that i can get the proper count, since it is pivot longered)
total_AD_alleles <- expanded_df %>% filter(Category == "AD") %>% nrow()
total_AD_alleles <- (total_AD_alleles/2)
total_CTRL_alleles <- expanded_df %>% filter(Category == "Control") %>% nrow()
total_CTRL_alleles <- (total_CTRL_alleles/2)

# Filter for LongHex alleles
long_VNTR_df <- expanded_df %>%
  filter(allele_type == "long_VNTR")

######################
# Count occurrences of each category
category_counts <- long_VNTR_df %>%
  group_by(Category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = ifelse(Category == "AD", (count / total_AD_alleles) * 100,
                             (count / total_CTRL_alleles) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", total_AD_alleles, total_CTRL_alleles)))  # Create a fraction string

# Create a contingency table for the chi-squared test
contingency_table <- table(long_hex_df$Category)

# Perform the chi-squared test
chi_squared_result <- chisq.test(contingency_table)

# Print the results
# print(contingency_table)
# print(chi_squared_result)

# Create the bar chart
ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
            position = position_stack(vjust = 0.5), 
            color = "black") +  # Add percentages and fractions on top of the bars
  labs(title = "Count of AD/CTRL alleles in all expanded VNTR alleles", x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
  theme_minimal() +
  scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))
```



# Are short or long VNTRs more associated with AD?:
```{r, include=F}

# total samples
ad_samples <- (alleleCall_filtered %>% filter(Category == "AD"))
total_ad_samples <- nrow(alleleCall_filtered %>% filter(Category == "AD"))

ctrl_samples <- (alleleCall_filtered %>% filter(Category == "Control"))
total_ctrl_samples <- nrow(alleleCall_filtered %>% filter(Category == "Control"))

long_VNTR_ad <- nrow(ad_samples %>% filter(allele1_VNTR == "long_VNTR"))
long_VNTR_ctrl <- nrow(ctrl_samples %>% filter(allele1_VNTR == "long_VNTR"))

perc_long_VNTR_ad <- long_VNTR_ad/total_ad_samples
perc_long_VNTR_ctrl <- long_VNTR_ctrl/total_ctrl_samples

data_to_plot <- data.frame(
  Category = c("AD", "Control"),
  Percentage = c(perc_long_VNTR_ad, perc_long_VNTR_ctrl)
)

# Create the plot with percentage labels
ggplot(data_to_plot, aes(x = Category, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = scales::percent(Percentage)), 
            vjust = -0.5,  # Adjust the vertical position of the text
            size = 3.7) +    # Adjust the size of the text
  scale_y_continuous(labels = scales::percent) +  # Format y-axis as percentage
  labs(title = "Percentage of Long VNTR Alleles by Category",
       x = "Category",
       y = "Percentage") +
  theme_minimal()

#making df with only important info for this:
longVNTR_filtereDF <- alleleCall_filtered %>% select(allele1_VNTR, Category)

# significantly different groups?
# Create a contingency table
contingency_table <- longVNTR_filtereDF %>%
  group_by(allele1_VNTR, Category) %>%
  summarise(count = n(),.groups = 'drop') %>%
  tidyr::pivot_wider(names_from = Category, values_from = count, values_fill = list(count = 0))

# View the contingency table
print(contingency_table)

# Perform Chi-squared test
chi_squared_result <- chisq.test(contingency_table[, -1])  # Exclude the allele1_hex column

# Print the results
print(chi_squared_result)
```

## Variation in hexamer types
# hex-hexvar types most associated with AD:
```{r}
#make new df for hex_hexvar 
hex_hexvar_df <- alleleCall_filtered %>% select(filename, allele1_hexhexvar, allele2_hexhexvar, Category, pattern)
#pivot longer the data so one allele is on each row
hex_hexvar_expanded_df <- hex_hexvar_df %>% 
  pivot_longer(cols = c(allele1_hexhexvar, allele2_hexhexvar), names_to = "allele", values_to = "hex_hexvar")
#counts of occurences
hex_hexvar_counts <- hex_hexvar_expanded_df %>%
  group_by(Category, hex_hexvar) %>%
  summarise(count = n(), .groups = 'drop') %>% 
  mutate(percentage = ifelse(Category == "AD", (count / total_AD_alleles) * 100,
                             (count / total_CTRL_alleles) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", total_AD_alleles, total_CTRL_alleles)))


# Create a contingency table
contingency_table_hex_hexvar <- hex_hexvar_counts %>%
  pivot_wider(names_from = hex_hexvar, values_from = count, values_fill = 0)

# Run Chi-squared test
# chi_squared_result_hex_hexvar <- chisq.test(contingency_table_hex_hexvar[, -1])  # Exclude the Category column

# Print the contingency table and Chi-squared test results
print(contingency_table_hex_hexvar)
# print(chi_squared_result_hex_hexvar)

# GRAPH:
total_count <- sum(hex_hexvar_counts$count)
hex_hexvar_counts$percentage_count <- hex_hexvar_counts$count / total_count

ggplot(hex_hexvar_counts, aes(x = hex_hexvar, y = percentage, fill = Category)) +
  geom_bar(stat = "identity", position = "dodge") +
  labs(title = "Hex-Hexvar Counts by Category",
       x = "Hex-Hexvar",
       y = "Percentage") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust = 1)) +
  geom_text(aes(label = paste0(round(percentage, 1), "%\n(", fraction, ")"), 
                        hjust = ifelse(Category == "AD", .5, .4)), 
            position = position_dodge(width = 0.9),  # Use position_dodge for alignment
            color = "black", 
            size = 3) 


```


#  hex-vntr-hexvar types most associated with AD
```{r}
longHex_filtereDF <- alleleCall_filtered %>% select(filename, allele1_hex, allele2_hex, allele1_VNTR, allele2_VNTR,Category, allele1_full_type, allele2_full_type)
# significantly different groups?

##########################################################
# Expand the dataframe to treat each allele as a separate sample
expanded_df <- longHex_filtereDF %>%
  pivot_longer(cols = c(allele1_hex, allele2_hex, allele1_VNTR, allele2_VNTR), names_to = "allele_position", values_to = "allele_type") %>% 
  pivot_longer(cols = c(allele1_full_type, allele2_full_type), values_to ="hexvar")
# Filter for LongHex alleles
long_hex_df <- expanded_df %>%
  filter(allele_type == "LongHex")


############################
# Count occurrences of hexvar for both AD and Control categories
hexvar_counts <- long_hex_df %>%
  group_by(Category, hexvar) %>%
  summarise(count = n(), .groups = 'drop') %>% 
  mutate(percentage = ifelse(Category == "AD", (count / total_AD_alleles) * 100,
                             (count / total_CTRL_alleles) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", total_AD_alleles, total_CTRL_alleles)))

# Create a contingency table
contingency_table <- hexvar_counts %>%
  pivot_wider(names_from = hexvar, values_from = count, values_fill = 0)

# Run Chi-squared test
# chi_squared_result <- chisq.test(contingency_table[, -1])  # Exclude the Category column

# Print the contingency table and Chi-squared test results
print(contingency_table)
print(chi_squared_result)

# Optional: Create a bar chart to visualize the results
# ggplot(hexvar_counts, aes(x = hexvar, y = count, fill = Category)) +
#   geom_bar(stat = "identity", position = "dodge"+  
  # scale_y_continuous(labels = scales::percent_format(accuracy = 1))) +
#   labs(title = "Hex-VNTR-Hexvar Counts by Category",
#        x = "Hexvar",
#        y = "Count") +
#   theme_minimal() +
#   geom_text(aes(label = count),  # Show raw counts instead of percentages
#             position = position_dodge(width = 0.9), 
#             vjust = -0.5, color = "black") +  # Adjust position above the bars
#   theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1))
###############################
# GRAPH:
total_count <- sum(hexvar_counts$count)
hexvar_counts$percentage_count <- hexvar_counts$count / total_count

ggplot(hexvar_counts, aes(x = hexvar, y = percentage, fill = Category)) +
  geom_bar(stat = "identity", position = "dodge") +
  labs(title = "Hex-VNTR-Hexvar Counts by Category",
       x = "Hex-VNTR-Hexvar",
       y = "Percentage") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust = 1)) +
  geom_text(aes(label = paste0(round(percentage, 1), "%\n(", fraction, ")"), 
                        hjust = ifelse(Category == "AD", .5, .4)), 
            position = position_dodge(width = 0.9),  # Use position_dodge for alignment
            color = "black", 
            size = 2.4) 
```

# homozygous LHex-hexv cases:
```{r}
#make new df for hex_hexvar 
hex_hexvar_df <- alleleCall_filtered %>% select(filename, allele1_hexhexvar, allele2_hexhexvar, Category, pattern)

#filtering hex_hexvar_df so that it only contains homozygous Lhex samples:
homozygous_LHex <- hex_hexvar_df %>%
  filter(startsWith(allele1_hexhexvar, paste0("LHex")) & startsWith(allele2_hexhexvar, paste0("LHex")))

# ggplot(data=homozygous_LHex,aes(x = allele1_hexhexvar, fill = Category))+
  # geom_bar(position = "dodge")

#make a counts table so that I can calculate % = (AD/totalAD)
category_counts <- homozygous_LHex %>%
  group_by(Category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = ifelse(Category == "AD", (count / 50) * 100,
                             (count / 27) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", 50, 27)))  # Create a fraction string

#graph:
# Create the bar chart
p <- ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
            position = position_stack(vjust = 0.5), 
            color = "black") +  # Add percentages and fractions on top of the bars
  labs(title = paste0("Percent of AD/CTRL alleles in all LHex-hexv/LHex-hexv Homozygous samples"), x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
  theme_minimal() +
  scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))

print(p)



############
#start of with LHex-hexv1 and other hexvar cases:
############
for (i in 0:6){
  #filtering hex_hexvar_df so that it only contains homozygous Lhex samples:
  homozygous_LHex <- hex_hexvar_df %>%
    filter(startsWith(allele1_hexhexvar, paste0("LHex_hexv", i)) & startsWith(allele2_hexhexvar, paste0("LHex_hexv", i)))
 
  # ggplot(data=homozygous_LHex,aes(x = allele1_hexhexvar, fill = Category))+
    # geom_bar(position = "dodge")
  
  #make a counts table so that I can calculate % = (AD/totalAD)
  category_counts <- homozygous_LHex %>%
    group_by(Category) %>%
    summarise(count = n(), .groups = 'drop') %>%
    mutate(percentage = ifelse(Category == "AD", (count / 50) * 100,
                               (count / 27) * 100),
           fraction = paste0(count, "/", ifelse(Category == "AD", 50, 27)))  # Create a fraction string
  
  #graph:
  # Create the bar chart
  p <- ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
    geom_bar(stat = "identity") +
    geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
              position = position_stack(vjust = 0.5), 
              color = "black") +  # Add percentages and fractions on top of the bars
    labs(title = paste0("Percent of AD/CTRL alleles in all LHex-hexv",i, " Homozygous samples"), x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
    theme_minimal() +
    scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))
  
  print(p)
}

```


# Looking at samples with only one sample that is long, and their likelihood of relation to AD, similar to above ^
```{r}
#make new df for hex_hexvar 
hex_hexvar_df <- alleleCall_filtered %>% select(filename, allele1_hexhexvar, allele2_hexhexvar, Category, pattern)

#filtering hex_hexvar_df so that it only contains homozygous Lhex samples:
heterozygous_LHex <- hex_hexvar_df %>%
  filter(startsWith(allele1_hexhexvar, paste0("SHex")) & startsWith(allele2_hexhexvar, paste0("LHex")))

# ggplot(data=homozygous_LHex,aes(x = allele1_hexhexvar, fill = Category))+
  # geom_bar(position = "dodge")

#make a counts table so that I can calculate % = (AD/totalAD)
category_counts <- heterozygous_LHex %>%
  group_by(Category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = ifelse(Category == "AD", (count / 50) * 100,
                             (count / 27) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", 50, 27)))  # Create a fraction string

#graph:
# Create the bar chart
p <- ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
            position = position_stack(vjust = 0.5), 
            color = "black") +  # Add percentages and fractions on top of the bars
  labs(title = paste0("Percent of AD/CTRL alleles in all SHex-hexv/LHex-hexv Heterozygous samples"), x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
  theme_minimal() +
  scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))

print(p)

##########
for (i in 0:6){
#filtering hex_hexvar_df so that it only contains one Lhex allele per samples:
  heterozygous_LHex <- hex_hexvar_df %>%
    filter(startsWith(allele1_hexhexvar, paste0("SHex_hexv", i)) & startsWith(allele2_hexhexvar, paste0("LHex_hexv", i)))
  
 # make a counts table so that I can calculate % = (AD/totalAD)
  category_counts <- heterozygous_LHex %>%
    group_by(Category) %>%
    summarise(count = n(), .groups = 'drop') %>%
    mutate(percentage = ifelse(Category == "AD", (count / 50) * 100,
                               (count / 27) * 100),
           fraction = paste0(count, "/", ifelse(Category == "AD", 50, 27)))  # Create a fraction string
  
  #graph:
  # Create the bar chart
  p <- ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
    geom_bar(stat = "identity") +
    geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
              position = position_stack(vjust = 0.5), 
              color = "black") +  # Add percentages and fractions on top of the bars
    labs(title = paste0("Percent of AD/CTRL alleles in all SHex-hexv",i, "/LHex-hexv",i," Heterozygous samples"), x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
    theme_minimal() +
    scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))
  
  print(p)
}

```

# homozygous SHex-hexv cases:
```{r}
#make new df for hex_hexvar 
hex_hexvar_df <- alleleCall_filtered %>% select(filename, allele1_hexhexvar, allele2_hexhexvar, Category, pattern)

#filtering hex_hexvar_df so that it only contains homozygous Shex samples:
homozygous_SHex <- hex_hexvar_df %>%
  filter(startsWith(allele1_hexhexvar, paste0("SHex")) & startsWith(allele2_hexhexvar, paste0("SHex")))

#make a counts table so that I can calculate % = (AD/totalAD)
category_counts <- homozygous_SHex %>%
  group_by(Category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(percentage = ifelse(Category == "AD", (count / 50) * 100,
                             (count / 27) * 100),
         fraction = paste0(count, "/", ifelse(Category == "AD", 50, 27)))  # Create a fraction string

#graph:
# Create the bar chart
p <- ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
            position = position_stack(vjust = 0.5), 
            color = "black") +  # Add percentages and fractions on top of the bars
  labs(title = paste0("Percent of AD/CTRL alleles in all SHex-hexv/SHex-hexv Homozygous samples"), x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
  theme_minimal() +
  scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))

print(p)



############
#start of with SHex-hexv1 and other hexvar cases:
############
for (i in 0:6){
  #filtering hex_hexvar_df so that it only contains homozygous Lhex samples:
  homozygous_SHex <- hex_hexvar_df %>%
    filter(startsWith(allele1_hexhexvar, paste0("SHex_hexv", i)) & startsWith(allele2_hexhexvar, paste0("SHex_hexv", i)))

  
  #make a counts table so that I can calculate % = (AD/totalAD)
  category_counts <- homozygous_SHex %>%
    group_by(Category) %>%
    summarise(count = n(), .groups = 'drop') %>%
    mutate(percentage = ifelse(Category == "AD", (count / 50) * 100,
                               (count / 27) * 100),
           fraction = paste0(count, "/", ifelse(Category == "AD", 50, 27)))  # Create a fraction string
  
  #graph:
  # Create the bar chart
  p <- ggplot(category_counts, aes(x = Category, y = percentage, fill = Category)) +
    geom_bar(stat = "identity") +
    geom_text(aes(label = paste0(round(percentage, 1), "% (", fraction, ")")), 
              position = position_stack(vjust = 0.5), 
              color = "black") +  # Add percentages and fractions on top of the bars
    labs(title = paste0("Percent of AD/CTRL alleles in all SHex-hexv",i, " Homozygous  SHex samples"), x = "Category", y = "Percentage") +  # Change y-axis label to "Count"
    theme_minimal() +
    scale_fill_manual(values = c("AD" = "#FFC20A", "Control" = "#CE38EC"))
  
  print(p)
}
```

